### DISCUSSION TITLE
Question about the design

### DISCUSSION THREAD

**Discussion by juanma1331** [bladewindui/ui#25] [Q&A] [answered]
Hi @mkocansey, first of all, good job!!.
I'm using your package to learn how to develop blade components properly.
I wanted to know the reasons behind placing the colors definition array on a prop instead of a variable on the php block.
The coloring prop on the button for example.
Thanks!! and keep going :)

**Comment by mkocansey** [comment, marked as answer]
Hey @juanma1331 thanks. I am glad you are picking some knowledge from the Bladewind components. 

To be honest placing the colours array at the props level was more of a personal preference. I prefer to keep attribute declarations at the props level and actual logic after the props declaration. So in this specific case, I consider `$coloring` to be a computed prop attribute that is based on the value of `$color`. 

However, it is perfectly possible to have declared `$coloring` within the `@php` block. You'd end up with the same results.

**Reply by juanma1331** [reply]
Really thanks, all clear!
